### DISCUSSION TITLE
Welcome to solar-logger Discussions!

### DISCUSSION THREAD

**Discussion by basking-in-the-sun2000** [basking-in-the-sun2000/solar-logger#8] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by PFema** [comment]
Hello! I am very inspired by your project and started my own, unfortunately, there is a problem with the registers for me. How did you retrieve those register? I found this document from huawei :

https://support.huawei.com/view/PdfRead/EDOC1100050690/SUPE_DOC/6001/document.pdf

but none of the registers listed there are the same as yours. Moreover, I can't seem to find anywhere on the datasheet that you can use ModBus TCP with a sun2000. Only RS485 (Modbus RTU) is available for communication. Would it be possible to give me a clearer idea on how you did this?

thanks in advance

**Reply by basking-in-the-sun2000** [reply]
The version I'm using is Interface Definitions (V3.0) for the Modbus registers. Also filled some of the blanks with the scanner files from the project. However, you can change the Huawei.py file to suit your needs, or even make a new one

Could be that your inverter doesn't include tcp, Huawei has a bunch of different versions. The residential models seem to do. Are you using a commercial one? 

Think it should be able to change the tcp calls to rtf. The library does both

**Comment by nocturno66** [comment]
Hi basking-in-the-sun2000
Thanks for your project.
I'm trying to execute in Visual Code for Windows and I get the following error:
PS C:\Users\nocturno\Dropbox\Micropic\Proyectos\solar-logger-master> & C:/Python39/python.exe c:/Users/nocturno/Dropbox/Micropic/Proyectos/solar-logger-master/main.py
Traceback (most recent call last):
  File "c:\Users\nocturno\Dropbox\Micropic\Proyectos\solar-logger-master\main.py", line 6, in <module>
    import emails
  File "c:\Users\nocturno\Dropbox\Micropic\Proyectos\solar-logger-master\emails.py", line 4, in <module>
    import config
  File "c:\Users\nocturno\Dropbox\Micropic\Proyectos\solar-logger-master\config.py", line 33, in <module>
    latitude = lat
NameError: name 'lat' is not defined

I have defined my latitude and longitude in data_sources.json but I get the same error.
Can you help me?

Thanks you very much

**Reply by basking-in-the-sun2000** [reply]
There are a series values you need to replace in the config.py file to reflect your site. Among these are the latitude and longitude (to actual numbers for your location). The data_sources.json is just the grafana definition for your databases. 

Try reading the README.md and setting_up.md. Let me know if these need further work to make them easier to understand.

**Reply by nocturno66** [reply]
I got it!!!!
![dashboard solar](https://user-images.githubusercontent.com/2972506/137289455-1a4a8872-d2dc-44f8-8a47-70131e7731d8.jpg)

Thanks you very much for your help

**Comment by basking-in-the-sun2000** [comment]
Great, glad you got it working

The colors might be off, since I set the thresholds according to my system levels. You might want to take a look at them, in the panels. Will try to take a look if these can be set as grafana variables, to make it easier if you upgrade the dashboards.

**Reply by nocturno66** [reply]
Yes, now it's time to tune it and put it to my liking.
Thanks

**Reply by nocturno66** [reply]
Hello again,
I am trying to integrate the application with Supla. I already have my token and have put it in supla_api.
But I don't know what to put in supla_dev_id and therefore it gives me an error.
I've been looking at supla.org and can't find a way to add the device to get the id.
Thanks in advance

**Reply by bdkacz** [reply]
Log in to supla account. click on your MEW-01 and You   have ID. Not ID from "My supla", but ID visible  after click on MEW-01 device :)

**Reply by nocturno66** [reply]
My MEW-01?
I don't see any device on my Supla account.

![supla1](https://user-images.githubusercontent.com/2972506/138542820-fd86b020-8fa6-471a-a60e-e5adf2c361e7.jpg)

![supla2](https://user-images.githubusercontent.com/2972506/138542822-d25d1998-fcb4-4465-9f5f-26a9056403b4.jpg)

Thanks

**Reply by bdkacz** [reply]
Do You add your MEW-01 into supla ? Do You see it in supla app ?

**Reply by nocturno66** [reply]
I don't have a MEW-01; I have a Raspberry Pi with this solar-logger firmware.
In Supla, I have looked for new devices and did not find any.
Thanks

**Reply by bdkacz** [reply]
Supła integration is for reading data from this device https://supla.zamel.com/en/product/mew-01/ If You don't have this device You can skip supła configuration

**Reply by nocturno66** [reply]
Ah, ok, I didn't know. I believed that the integration was with the Pi.
Thank you

**Comment by ScApi** [comment]
I'm trying to properly configure Supla part but no matter what I try most of the times i get
supla error: Invalid base64-encoded string: number of data characters (5) cannot be 1 more than a multiple of 4
It's probably something trivial but I'm missing it.

Edit: I've managed to get it sorted out (need to authorize token) but I don't see any movement in dashboard.
GRID stays at 0

**Reply by basking-in-the-sun2000** [reply]
Sorry, I really never looked into the supla code, since @bdkacz provided it.
